tests/zfs-tests/cmd/largest_file/largest_file.c

```diff
@@ -79,8 +79,8 @@ main(int argc, char **argv)
 	fd = open(testfile, O_CREAT | O_RDWR, mode);
 	if (fd < 0) {
 		perror("Failed to create testfile");
-		err = errno;
-		goto out;
+		free(testfile);
+		return errno;
```

Comment:
This isn't quite right.  `errno` may be changed by the call the `free()`.  In fact, errno may be changed by `perror` and it should be saved before it's called, see `perror(3)`.  Also there's a cstyle issue regarding return, it should be `return (errno)`.  This first block should be changed to:

```
		err = errno;
		perror(...);
		free(testfile);
		return (err);

```

And the subsequent goto blocks should be:

```
		err = errno;
		perror(...);
		goto out;
```